prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -213,31 +213,45 @@ void rxr_pkt_init_req_hdr(struct rxr_ep *ep,
 	assert(opt_hdr - pkt_entry->pkt == rxr_pkt_req_hdr_size(pkt_entry));
 }
 
+/**
+ * @brief Get the value of rma_iov_count in pkt header.
+ *
+ * @param[in] pkt_entry the pkt entry
+ * @return The rma_iov_count in the pkt header, if it is part of the header.
+ * Otherwise return 0.
+ */
+uint32_t rxr_pkt_hdr_rma_iov_count(struct rxr_pkt_entry *pkt_entry)
+{
+	int pkt_type = rxr_get_base_hdr(pkt_entry->pkt)->type;
+	if (pkt_type == RXR_EAGER_RTW_PKT ||
+	    pkt_type == RXR_DC_EAGER_RTW_PKT ||
+	    pkt_type == RXR_LONGCTS_RTW_PKT ||
+	    pkt_type == RXR_DC_LONGCTS_RTW_PKT ||
+	    pkt_type == RXR_LONGREAD_RTW_PKT)
+		return rxr_get_rtw_base_hdr(pkt_entry->pkt)->rma_iov_count;
+	else if (pkt_type == RXR_SHORT_RTR_PKT ||
```

Comment:
`else` is not necessary.

Reply:
Thank you! Fixed.

Reply:
I still see `else`. Do you forget to push?

Reply:
Sorry, I misunderstood your earlier comment. I thought you only wanted to remove the last `else`. Now I remove all the `else`. Please see the latest revision.